Marlin/Marlin_main.cpp

```diff
@@ -2121,28 +2115,18 @@ static void clean_up_after_endstop_or_probe_move() {
       planner.bed_level_matrix.set_to_identity();
     #endif
 
-    #if ENABLED(DELTA)
-      float z_before = current_position[Z_AXIS],         // Current Z
-            z_mm = stepper.get_axis_position_mm(Z_AXIS); // Some tower's current position
-    #endif
-
     do_blocking_move_to_z(-(Z_MAX_LENGTH + 10), Z_PROBE_SPEED_FAST);
     endstops.hit_on_purpose();
-    SET_Z_FROM_STEPPERS();
+    set_current_from_steppers();
```

Comment:
Of course there is always a single-axis option. It would do redundant computation if used to set 3 axes, but it's simple enough…
```cpp
void get_axis_pos_from_steppers(AxisEnum axis) {
  #if ENABLED(DELTA)
    set_cartesian_from_steppers();
    return LOGICAL_POSITION(cartesian_position[axis], axis);
  #elif ENABLED(AUTO_BED_LEVELING_FEATURE)
    vector_3 pos = planner.adjusted_position();
    return LOGICAL_POSITION(axis == X_AXIS ? pos.x : axis == Y_AXIS ? pos.y : pos.z, axis);
  #else
    return LOGICAL_POSITION(stepper.get_axis_position_mm(axis), axis); // CORE handled transparently
  #endif
}
```